compiler/rustc_transmute/src/layout/tree.rs

```diff
@@ -503,13 +495,15 @@ pub(crate) mod rustc {
         (ty, layout): (Ty<'tcx>, Layout<'tcx>),
         i: FieldIdx,
     ) -> Ty<'tcx> {
+        // FIXME: Why does this not just use `ty_and_layout_field`?
```

Comment:
@jswrenn would be nice to have a short comment here on why this seemingly duplicates the `ty_and_layout_field` logic.

Reply:
It's to avoid the call to `layout_of` in `ty_and_layout_field`, which, IIRC, either requires that regions are erased with placeholders, or itself erases the regions with placeholders. (I think it was the former?) Regardless, we don't want to erase regions in field types since they might need to be handed back to selection as outlives obligations.

I can dig into this a bit tomorrow and get back to you with a prepared comment.

Reply:
Oh yeah, the entire layout machinery works on erased types. You'll have a hard time preserving lifetime information there.

Long-term however we also really don't want to maintain two versions of `ty_and_layout_field` each with their own distinct set of bugs...

Reply:
Confirmed that `ty_and_layout_field` replaces regions with placeholders.
```suggestion
        // We cannot use `ty_and_layout_field` to retrieve the field type, since
        // `ty_and_layout_field` erases regions in the returned type. We must
        // not erase regions here, since we may need to ultimately emit outlives
        // bounds as a consequence of the transmutability analysis.
```